kolibri/core/api.py

```diff
@@ -45,3 +53,22 @@ def finalize_response(self, request, response, *args, **kwargs):
             "attachment; filename=%s.csv" % self.csv_export_filename
         )
         return response
+
+
+class PortalViewSet(viewsets.ViewSet):
+
+    @action(detail=False, methods=["post"])
+    def register(self, request):
+        facility = Facility.objects.get(id=request.data.get('facility_id'))
+        try:
+            response = registerfacility(request.data.get('token'), facility)
+        except requests.exceptions.RequestException as e:  # bubble up any response error
+            return Response(e.response.json(), status=e.response.status_code)
+        return Response(status=response.status_code)
+
+    @action(detail=False, methods=["get"])
+    def validate_token(self, request):
+        PORTAL_URL = conf.OPTIONS["Urls"]["DATA_PORTAL_SYNCING_BASE_URL"]
+        # token is in query params
+        response = requests.get(urljoin(PORTAL_URL, 'portal/api/public/v1/registerfacility/validate_token'), params=request.query_params)
+        return Response(response.json(), status=response.status_code)
```

Comment:
I tried registering the token `test` and got this error:

```
ERROR    Internal Server Error: /api/auth/portal/validate_token/
Traceback (most recent call last):
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/rest_framework/viewsets.py", line 122, in view
    return self.dispatch(request, *args, **kwargs)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/rest_framework/views.py", line 496, in dispatch
    response = self.handle_exception(exc)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/rest_framework/views.py", line 455, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/rest_framework/views.py", line 493, in dispatch
    response = handler(request, *args, **kwargs)
  File "/Users/d/Projects/le/kolibri/kolibri/core/api.py", line 74, in validate_token
    return Response(response.json(), status=response.status_code)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/requests/models.py", line 978, in json
    return complexjson.loads(self.text, **kwargs)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/simplejson/__init__.py", line 518, in loads
    return _default_decoder.decode(s)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/simplejson/decoder.py", line 370, in decode
    obj, end = self.raw_decode(s)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/simplejson/decoder.py", line 400, in raw_decode
    return self.scan_once(s, idx=_w(s, idx).end())
simplejson.errors.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
ERROR    "GET /api/auth/portal/validate_token/?token=test&1573071325456=1573071325456 HTTP/1.1" 500 32392
```

Reply:
yeah that's because the PR on kdp that supports this hasn't been merged